pennylane/resources/first_quantization.py

```diff
@@ -15,505 +15,559 @@
 This module contains the functions needed for estimating the number of logical qubits and
 non-Clifford gates for quantum algorithms in first quantization using a plane-wave basis.
 """
-# pylint: disable=too-many-arguments
+# pylint: disable=no-self-use disable=too-many-arguments disable=too-many-instance-attributes
 from scipy import integrate
 from pennylane import numpy as np
+from pennylane.operation import Operation, AnyWires
 
 
-def success_prob(n, br):
-    r"""Return the probability of success for state preparation.
+class FirstQuantization(Operation):
+    r"""Contains the functionality for estimating the number of non-Clifford gates and logical qubits
+    for quantum algorithms in first quantization using a plane-wave basis.
```

Comment:
```suggestion
    r"""Estimates the number of non-Clifford gates and logical qubits
    for quantum algorithms in first quantization using a plane-wave basis.
```
We should probably also shorten the description for `DoubleFactorization`. Looking at the built docs, they seem too clunky to me